### ISSUE TITLE
thermal-mesh: JAX-FEM tesseract applies the Neumann flux with a sign inconsistent with its source term

### ISSUE THREAD

**Issue by meyer-nils** [pasteurlabs/mosaic#107]
First off, mosaic is a great initiative, and thanks for including torch-fem! While double-checking the torch-fem results I noticed something in the thermal-mesh source experiments that I believe is a small sign bug in the JAX-FEM tesseract, not a real solver disagreement.

## Observation

In `forward/source_baseline` and `forward/source_linearity`, torch-fem and JAX-FEM are exact mirror images. The physics is linear, so the compliance must be affine in the source amplitude $a$, and both curves are with the same intercept and slope magnitude, but opposite slope signs:

- torch-fem: $C(a) = 15.889 + 1.975a$
- JAX-FEM: $C(a) = 15.889 - 1.975a$

The intercept is the flux-only compliance (analytically $Q^2 L_x / (\kappa L_y L_z) = 15.8888$ at uniform $\rho=0.5$). The disagreement is purely the sign of the flux × source cross term $a \oint q_n T_s\, d\Gamma$. A positive heat source warms the domain, so I believe the flux face should get hotter and $C$ must grow. 

## Suggested fix

JAX-FEM adds surface maps to the weak-form residual, so a prescribed influx must enter negated (compare the surface map in JAX-FEM's own Poisson demo). In [`mosaic/tesseracts/thermal-mesh/jax-fem/tesseract_api.py#L240`](https://github.com/pasteurlabs/mosaic/blob/639a1a3/mosaic/tesseracts/thermal-mesh/jax-fem/tesseract_api.py#L240):

```python
def make_neumann_value_fn(i: int) -> Callable:
    def _value_fn_vn(u: jnp.ndarray, x: jnp.ndarray):
        return von_neumann_values[i]   # ← should be: return -von_neumann_values[i]
```

Currently the solve applies the flux as an outflux, and `compute_thermal_compliance` compensates with `flux = -jax.vmap(...)` (line 174). The double negation cancels in the flux-only experiments (baseline, agreement, and the ρ-gradient checks all look fine) but it flips the cross term whenever a volumetric source is active. With the one-line fix above, line 174 then computes the physical inward flux and can stay exactly as it is. As a side benefit, the temperature field aligns with the `_approx_target_temperature` targets, so the source-identification experiments become meaningful for JAX-FEM too.

Happy to open a PR with the one-line change if that's easier. Thanks again for the careful work on this benchmark!

## Disclaimer
I am the author of torch-fem, so I might be biased in thinking that the torch-fem result is the correct one. So please double check. 😀

**Comment by dionhaefner** [comment]
Confirmed, thanks! Fix incoming.
